src/main/java/rx/swing/sources/AbstractButtonSource.java

```diff
@@ -44,14 +42,14 @@ public void actionPerformed(ActionEvent e) {
                     }
                 };
                 button.addActionListener(listener);
-                subscriber.add(SwingSubscriptions.unsubscribeInEventDispatchThread(new Action0() {
+                subscriber.add(Subscriptions.create(new Action0() {
                     @Override
                     public void call() {
                         button.removeActionListener(listener);
                     }
                 }));
             }
-        });
+        }).subscribeOn(SwingScheduler.getInstance());
```

Comment:
@zsxwing Yikes, my mistake. You even mentioned it in the original issue and I just forgot. I'll do a new pr.

Something like this?
```
    public static Observable<ActionEvent> fromActionOf(final AbstractButton button) {
        return Observable.create(new OnSubscribe<ActionEvent>() {
            @Override
            public void call(final Subscriber<? super ActionEvent> subscriber) {
                final ActionListener listener = new ActionListener() {
                    @Override
                    public void actionPerformed(ActionEvent e) {
                        subscriber.onNext(e);
                    }
                };
                button.addActionListener(listener);
                subscriber.add(Subscriptions.create(new Action0() {
                    @Override
                    public void call() {
                        button.removeActionListener(listener);
                    }
                }));
            }
        }).subscribeOn(SwingScheduler.getInstance())
          .unsubscribeOn(SwingScheduler.getInstance());
    }
```